### PR TITLE
refactor: Convert `openmct-yamcs` to ESModule

### LINKED ISSUE #409: [CI] CI is broken after the conversion to modules
#### Summary
Open MCT was converted to a module. CI and tests are broken.

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #409 

### Describe your changes:

- We need to update openmct-yamcs to match the cjs import syntax for tests.
- Also, the tests changed in 3.3.0 and are updated with the new locators.
- Added lint step
- Added linting rules for ESModule Import
- Remove Deploysentinel
- Added e2e watch mode to match openmct
- Update network tests to inspect the request/response

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: tests/e2e/yamcs/search.e2e.spec.js [hunk 1]
```diff
@@ -34,13 +34,14 @@ test.describe("Quickstart search tests @yamcs", () => {
         await page.locator('[aria-label="OpenMCT Search"] [aria-label="Search Input"]').click();
         // Search for Sequence
         await page.locator('[aria-label="OpenMCT Search"] [aria-label="Search Input"]').fill('Sequence');
-        await expect(page.locator('[aria-label="Search Result"] >> nth=0')).toContainText("CCSDS_Packet_Sequence");
-        await expect(page.locator('[aria-label="Search Result"] >> nth=1')).toContainText("CCSDS_Packet_Sequence.GroupFlags");
-        await expect(page.locator('[aria-label="Search Result"] >> nth=2')).toContainText("CCSDS_Packet_Sequence.Count");
+
+        await expect(page.getByLabel('Object Search Result').nth(0)).toContainText("CCSDS_Packet_Sequence");
```
Comment (unlikelyzero) [inline tests/e2e/yamcs/search.e2e.spec.js:38]: test fixes to match 3.3.0

**Review by scottbell — COMMENTED** [review]
Tested this in my branch, and code is good, but still looks like some tests are still failing?

**Review by unlikelyzero — APPROVED** [review]

**Review by ozyx — DISMISSED** [review]
We can't merge this until we know consumers can still import it. I tested this with VIPER and it's totally borked, so likely (possibly?) a webpack configuration or something else we're missing.

**Comment by shefalijoshi** [comment]
> We can't merge this until we know consumers can still import it. I tested this with VIPER and it's totally borked, so likely (possibly?) a webpack configuration or something else we're missing.

OK I'll update with changes after investigation.

File: package.json [hunk 2]
```diff
@@ -19,6 +20,7 @@
     "test:e2e:smoke": "npx playwright test --config=./tests/e2e/playwright-quickstart.config.js --project=chromium quickstartSmoke",
     "test:e2e:quickstart": "npx playwright test --config=./tests/e2e/playwright-quickstart.config.js --project=chromium tests/e2e/yamcs/",
     "test:e2e:quickstart:local": "npx playwright test --config=./tests/e2e/playwright-quickstart.config.js --project=local-chrome tests/e2e/yamcs/",
+    "test:e2e:watch": "npx playwright test --ui --config=./tests/e2e/playwright-quickstart.config.js",
```
Comment (unlikelyzero) [inline package.json:23]: driveby to write tests faster

File: package.json [hunk 3]
```diff
@@ -36,11 +38,11 @@
   "devDependencies": {
     "@babel/core": "7.20.12",
     "@babel/eslint-parser": "7.19.1",
-    "@deploysentinel/playwright": "0.3.4",
```
Comment (unlikelyzero) [inline package.json:39]: driveby

File: package.json [hunk 4]
```diff
@@ -36,11 +38,11 @@
...
     "@playwright/test": "1.39.0",
     "babel-loader": "9.1.0",
     "babel-plugin-istanbul": "6.1.1",
     "eslint": "8.38.0",
+    "eslint-plugin-import":"2.29.1",
```
Comment (unlikelyzero) [inline package.json:45]: make sure we didn't miss anything

File: tests/e2e/yamcs/network.e2e.spec.js [hunk 5]
```diff
@@ -24,63 +24,86 @@
...
     let networkRequests = [];
     let filteredRequests = [];
+    let parameterArchiveGet;
+    let batchPost;
+    let allParams;
+    let userGet;
+    let mdbOverride;
 
-    test('Validate network traffic to YAMCS', async ({ page }) => {
+    test.only('Validate network traffic to YAMCS', async ({ page }) => {
```
Comment (ozyx) [inline tests/e2e/yamcs/network.e2e.spec.js:39]: `.only`

File: tests/e2e/yamcs/network.e2e.spec.js [hunk 6]
```diff
@@ -24,63 +24,86 @@
...
+        allParams = page.waitForResponse('/yamcs-proxy/api/mdb/myproject/parameters?details=yes&limit=1000');
+        userGet = page.waitForResponse('/yamcs-proxy/api/user/');
+        mdbOverride = page.waitForResponse('/yamcs-proxy/api/mdb-overrides/myproject/realtime');
+
+        // First Load Tests
         await page.goto("./", { waitUntil: "networkidle" });
 
+        await Promise.all([allParams, userGet, mdbOverride]);
+
+        //I'm not sure what this block does yet
```
Comment (ozyx) [inline tests/e2e/yamcs/network.e2e.spec.js:51]: 👁️

**Review by ozyx — CHANGES_REQUESTED** [review]
I found a rogue `.only()`

File: check-optional-dependencies.js [hunk 7]
```diff
@@ -1,14 +1,15 @@
 // check-optional-dependencies.js
-const fs = require('fs');
-const semver = require('semver');
-const myPackageJson = require('./package.json');
+import fs from 'node:fs';
```
Comment (scottbell) [inline check-optional-dependencies.js:2]: we should be using [fs/promises](https://nodejs.org/docs/latest-v18.x/api/fs.html#promise-example)

File: tests/e2e/yamcs/network.e2e.spec.js [hunk 8]
```diff
@@ -24,63 +24,86 @@
...
+        parameterArchiveGet = page.waitForResponse('**/api/archive/myproject/parameters/myproject/CCSDS_Packet_Length/samples**');
+        batchPost = page.waitForResponse('**/api/processors/myproject/realtime/parameters:batchGet');
+        await page.getByRole('treeitem', { name: 'CCSDS_Packet_Length' }).click()
 
-        // wait for debounced requests in YAMCS Latest Telemetry Provider to finish
-        await new Promise(resolve => setTimeout(resolve, 500));
+        await Promise.all([parameterArchiveGet, batchPost]);
 
+        await page.waitForLoadState('networkidle');
+        await new Promise(resolve => setTimeout(resolve, 500));
```
Comment (scottbell) [inline tests/e2e/yamcs/network.e2e.spec.js:82]: do we need this with the `waitForResponse` stuff?

File: tests/e2e/yamcs/network.e2e.spec.js [hunk 9]
```diff
@@ -106,23 +133,28 @@ test.describe("Quickstart network requests @yamcs", () => {
...
+        batchPost = page.waitForResponse('**/api/processors/myproject/realtime/parameters:batchGet');
         // 6. GET for telemetry item mdb overrides
+        mdbOverride = page.waitForResponse('/yamcs-proxy/api/mdb-overrides/myproject/realtime');
+
+        networkRequests = [];
+        await page.reload( {waitUntil : 'networkidle'});
+        await Promise.all([allParams, userGet, mdbOverride, parameterArchiveGet, batchPost, mdbOverride]);
+
+        // wait for debounced requests in YAMCS Latest Telemetry Provider to finish
+        await new Promise(resolve => setTimeout(resolve, 500));
```
Comment (scottbell) [inline tests/e2e/yamcs/network.e2e.spec.js:155]: same as above - is this still needed?

**Review by scottbell — CHANGES_REQUESTED** [review]

**Pull request merged**